### DISCUSSION TITLE
Should Clients be reused or created for each request?

### DISCUSSION THREAD

**Discussion by dmalmkvist** [deepgram/community#712] [General help] [answered]
I was wondering if there is a recommended pattern for using the go client, for example is it expected that a new client is created for each request or can the client be reused? Can the client handle multiple requests in parallel?

This is for both the `client.New()` and `prerecorded.New()` clients.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jpvajda** [comment, marked as answer]
@dmalmkvist  The Deepgram Go client can be reused for multiple requests and can handle multiple requests in parallel. It is not necessary to create a new client for each request.

**Reply by iliesh** [reply]
Is this valid for live transcription as well? For instance,if I create only a single Go client and send raw traffic to Deepgram without creating separate instances for each call, I will receive all transcriptions in a single callback function that was set when the client was created, how would I distunguish the transcription for each call then? Is it possible to tag each RTP packet with some sort of tags and then use them in the callback function?

**Reply by davidvonthenen** [reply]
You can reuse the Live Client, but remember there is a 1-to-1 correlation/mapping/etc with a single audio stream to a single websocket. This means you can't send multiple audio segments from different audio streams on the same Live Client.

**Comment by davidvonthenen** [comment]
@jpvajda is absolutely correct

You can reuse even in this fashion:
```
func main()
{
  	...
  	...
  	// create a Deepgram client
	c := client.NewWithDefaults()
  	...
  	...
  	// create an analyze client
	analyzeClient := analyze.New(c)
  	...
  	...
  	// create an manage client
	manageClient := manage.New(c)
  	...
  	...
}
```
